### PR TITLE
Waypoint creation/move/deletion

### PR DESCRIPTION
**Pull request by loic-fejoz**
Add `MeshInterface.sendWaypoint` to send (create or move) waypoint, and `MeshInterface.deleteWaypoint`  to delete one.
Add an example script to create, move, delete waypoint.

`MeshInterface.sendWaypoint`

Example scripts usage looks like:

``` bash
python3 examples/waypoint.py --port /dev/ttyUSB0 create 45 test "the description" "2024-12-18T23:05:23" 48.74 7.35
python3 examples/waypoint.py delete 45
```

### CODE REVIEW

**Review by ianmcorvidae — CHANGES_REQUESTED** [review]
Thanks for the pull request! I've got a few questions and some things that appear to be missing. Would love to see this added to the API!

File: meshtastic/mesh_interface.py [hunk 1]
```diff
@@ -700,6 +702,98 @@ def onResponseTelemetry(self, p: dict):
...
+            w.id = id
+        if latitude != 0.0:
+            w.latitude_i = int(latitude * 1e7)
+            logging.debug(f"w.latitude_i:{w.latitude_i}")
+        if longitude != 0.0:
+            w.longitude_i = int(longitude * 1e7)
+            logging.debug(f"w.longitude_i:{w.longitude_i}")
+
+        if wantResponse:
+            onResponse = self.onResponseWaypoint
```
[the remark this answers is not in the document]
Comment (loic-fejoz) [reply meshtastic/mesh_interface.py:757]: My bad, I forgot those indeed.
Latest commit add them but I am not sure how they should really behave from a protocol perspective. So please guide me if it is wrong.

File: meshtastic/mesh_interface.py [hunk 2]
```diff
@@ -700,6 +702,98 @@ def onResponseTelemetry(self, p: dict):
...
+            w,
+            destinationId,
+            portNum=portnums_pb2.PortNum.WAYPOINT_APP,
+            wantAck=wantAck,
+            wantResponse=wantResponse,
+            onResponse=onResponse,
+            channelIndex=channelIndex,
+        )
+        if wantResponse:
+            self.waitForWaypoint()
```
[the remark this answers is not in the document]
Comment (loic-fejoz) [reply meshtastic/mesh_interface.py:771]: see commit 7cc18e9df627a8730fdd1777789230c239bf5e0a

File: meshtastic/mesh_interface.py [hunk 3]
```diff
@@ -700,6 +702,98 @@ def onResponseTelemetry(self, p: dict):
...
+        Returns the sent packet. The id field will be populated in this packet and
+        can be used to track future message acks/naks.
+        """
+        w = mesh_pb2.Waypoint()
+        w.name = name
+        w.description = description
+        w.expire = expire
+        if id is None:
+            seed = secrets.randbits(32)
+            w.id = math.floor(seed * math.pow(2, -32) * 1e9)
```
[the remark this answers is not in the document]
Comment (loic-fejoz) [reply meshtastic/mesh_interface.py:745]: I have done similar has in [javascript API](https://github.com/meshtastic/js/blob/715e35d2374276a43ffa93c628e3710875d43907/src/meshDevice.ts#L791).
Note that it is not a packet ID but a waypoint ID.

[the remark this answers is not in the document]
Comment (ianmcorvidae) [reply meshtastic/mesh_interface.py:745]: Fair enough. I'm not sure why restricting it to 9 digits is useful, but it's probably fine in any case for this applicaiton.

File: meshtastic/mesh_interface.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ianmcorvidae) [reply meshtastic/mesh_interface.py:757]: These functions are what'd normally be used by the CLI if it supported this stuff, but of course this PR doesn't include that. For now, I think they can just serve as placeholders, so what you have should be fine, I think.

**Review by ianmcorvidae — APPROVED** [review]

**Comment by ianmcorvidae** [comment]
I'm just running the CI checks on this to see if anything pops up there. As long as that's all fine, this seems good to merge.

**Pull request merged**